coordinator/internal/stateguard/stateguard_test.go

```diff
@@ -35,13 +40,207 @@ contrast_coordinator_manifest_generation %d
 `
 )
 
+func TestUpdateState(t *testing.T) {
+	ctx := t.Context()
+	assert := assert.New(t)
+	require := require.New(t)
+	g, _ := newTestGuard(t)
+
+	// A new Guard is empty.
+	emptyState, err := g.GetState(ctx)
+	require.Nil(emptyState)
+	require.ErrorIs(err, ErrNoState)
+
+	mnfst, manifestBytes, policies := newManifest(t)
+
+	se := newSeedEngine(t)
+	updateState, err := g.UpdateState(ctx, emptyState, se, manifestBytes, policies)
+	require.NoError(err)
+	require.NotNil(updateState)
+
+	// GetState now returns the same state as UpdateState.
+	getState, err := g.GetState(ctx)
+	require.NoError(err)
+	require.NotNil(getState)
+	require.Equal(getState, updateState)
+
+	// The returned state contains the values passed to UpdateState.
+	assert.Equal(mnfst, getState.Manifest())
+	assert.Equal(manifestBytes, getState.ManifestBytes())
+	assert.Same(se, getState.SeedEngine())
+
+	// The state has a CA with valid certs.
+	ca := getState.CA()
+	require.NotNil(ca)
+	for name, pemBytes := range map[string][]byte{
+		"root CA":         ca.GetRootCACert(),
+		"intermediate CA": ca.GetIntermCACert(),
+		"mesh CA":         ca.GetMeshCACert(),
+	} {
+		pool := x509.NewCertPool()
+		assert.True(pool.AppendCertsFromPEM(pemBytes), name)
+		block, rest := pem.Decode(pemBytes)
+		assert.NotEmpty(block.Bytes, name)
+		assert.Empty(rest, name)
+	}
+}
+
+// TestUpdateStateRace tests that parallel changes to the internal state pointer don't affect the
+// outcome of UpdateState calls.
+func TestUpdateStateRace(t *testing.T) {
```

Comment:
I was confused by the test name here, I expected to see a race test (something trying to trigger the race detector by hammering with goroutines). I think the test makes sense, but maybe frame it more like a "background update" than parallel.